CoreQualityControlView/test/unit/src/au/gov/asd/tac/constellation/views/qualitycontrol/widget/DefaultQualityControlAutoButtonNGTest.java

```diff
@@ -0,0 +1,135 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.views.qualitycontrol.widget;
+
+import au.gov.asd.tac.constellation.graph.StoreGraph;
+import au.gov.asd.tac.constellation.graph.schema.Schema;
+import au.gov.asd.tac.constellation.graph.schema.SchemaFactoryUtilities;
+import au.gov.asd.tac.constellation.graph.schema.analytic.AnalyticSchemaFactory;
+import au.gov.asd.tac.constellation.views.qualitycontrol.QualityControlEvent;
+import au.gov.asd.tac.constellation.views.qualitycontrol.QualityControlEvent.QualityCategory;
+import au.gov.asd.tac.constellation.views.qualitycontrol.daemon.QualityControlState;
+import au.gov.asd.tac.constellation.views.qualitycontrol.rules.QualityControlRule;
+import java.awt.GraphicsEnvironment;
+import java.util.ArrayList;
+import java.util.List;
+import javafx.embed.swing.JFXPanel;
+import javafx.scene.control.Tooltip;
+import static org.mockito.Mockito.*;
+import static org.testng.Assert.assertEquals;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class DefaultQualityControlAutoButtonNGTest {
+
+    private StoreGraph graph;
+    private List<QualityControlEvent> events;
+    private List<QualityControlRule> rules;
+
+    //Dependencies (will be mocked)
+    private QualityControlEvent qualityControlEvent;
+
+    public DefaultQualityControlAutoButtonNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+        final Schema schema = SchemaFactoryUtilities.getSchemaFactory(AnalyticSchemaFactory.ANALYTIC_SCHEMA_ID).createSchema();
+        graph = new StoreGraph(schema);
+        rules = new ArrayList<>();
+        events = new ArrayList<>();
+
+        //mock QualityControlEvent
+        qualityControlEvent = mock(QualityControlEvent.class);
+        when(qualityControlEvent.getCategory()).thenReturn(QualityCategory.FATAL);
+        when(qualityControlEvent.getReasons()).thenReturn("Reason 1, Reason2");
+
+        events.add(qualityControlEvent);
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of qualityControlChanged method, of class
+     * DefaultQualityControlAutoButton, with Null State.
+     */
+    @Test
+    public void testQualityControlChangedWithNullState() throws InterruptedException {
+        System.out.println("qualityControlChanged");
+        QualityControlState state = null;
```

Comment:
I would remove this variable and just passed null directly to any objects or functions using it.